### ISSUE TITLE
[Bug]: crash on start

### ISSUE THREAD

**Issue by nvtkaszpir** [project-zot/zot#2407]
### zot version

v2.0.3-0-g819994c

### Describe the bug

Crash on start with trivy scan enabled.
I've got about 14GB of publicly available images only from various mirrors.

### To reproduce

1. Configuration
/etc/zot/config.json
```json
{
  "distSpecVersion": "1.1.0",
  "storage": {
    "dedupe": true,
    "gc": true,
    "gcDelay": "1h",
    "gcInterval": "2h",
    "rootDirectory": "/tank/zot"
  },
  "http": {
    "address": "0.0.0.0",
    "port": "5000"
  },
  "log": {
    "level": "debug",
    "output": "/var/log/zot/zot.log",
    "audit": "/var/log/zot/audit.log"
  },
  "extensions": {
    "ui": {
      "enable": true
    },
    "search": {
      "enable": true,
      "cve": {
        "updateInterval": "168h"
      }
    },
    "metrics": {
      "enable": true,
      "prometheus": {
        "path": "/metrics"
      }
    },
    "mgmt": {
      "enable": true
    },
    "scrub": {
      "enable": false,
      "interval": "24h"
    },
    "sync": {
      "enable": true,
      "registries": [
        {
          "urls": [
            "https://index.docker.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/docker.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true,
          "maxRetries": 3,
          "retryDelay": "15m"
        },
        {
          "urls": [
            "https://gcr.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/gcr.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://ghcr.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/ghcr.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://k8s.gcr.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/k8s.gcr.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://nvcr.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/nvcr.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://quay.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/quay.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://registry.gitlab.com"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/registry.gitlab.com"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://registry.k8s.io"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/registry.k8s.io"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        },
        {
          "urls": [
            "https://public.ecr.aws"
          ],
          "content": [
            {
              "prefix": "**",
              "destination": "/public.ecr.aws"
            }
          ],
          "onDemand": true,
          "tlsVerify": true
        }
      ]
    }
  }
}
```

2. Client tool used

4. Seen error
sudo -u zot zot serve /etc/zot/config.json
```text
2024-04-22T23:02:27.143+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.145+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.186+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.186+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.196+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.197+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.206+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.207+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.228+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.229+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.248+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.249+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.257+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.258+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.266+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.273+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.273+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.274+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.274+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.290+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.291+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.299+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.300+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:27.307+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.312+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.312+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.313+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.313+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.328+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.334+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.334+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.335+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.335+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.350+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.356+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.356+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.359+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.359+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.376+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.383+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.383+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.384+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.384+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.396+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.406+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.406+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.407+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.407+0200	INFO	Detecting node-pkg vulnerabilities...
2024-04-22T23:02:27.448+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.449+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.449+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.460+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.461+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.461+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.474+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.475+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.475+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.491+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.493+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.493+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.513+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.514+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.514+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.547+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.564+0200	INFO	Detected OS: debian
2024-04-22T23:02:27.564+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:27.579+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.579+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.627+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.637+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.637+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.642+0200	INFO	Number of language-specific files: 2
2024-04-22T23:02:27.642+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.652+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.15.8
2024-04-22T23:02:27.652+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:27.673+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.684+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.684+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.688+0200	INFO	Number of language-specific files: 2
2024-04-22T23:02:27.688+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.698+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.15.8
2024-04-22T23:02:27.698+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:27.717+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.736+0200	INFO	Detected OS: debian
2024-04-22T23:02:27.736+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:27.750+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.750+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.773+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.777+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.777+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.778+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.778+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.801+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.813+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.813+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.815+0200	INFO	Number of language-specific files: 2
2024-04-22T23:02:27.815+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.827+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.15.8
2024-04-22T23:02:27.827+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:27.842+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.868+0200	INFO	Detected OS: debian
2024-04-22T23:02:27.868+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:27.889+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.889+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.915+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.920+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.920+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.922+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.922+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.937+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.940+0200	INFO	Detected OS: alpine
2024-04-22T23:02:27.940+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:27.941+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.941+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:27.958+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:27.975+0200	INFO	Detected OS: debian
2024-04-22T23:02:27.976+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:27.991+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:27.991+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.023+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.040+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.040+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.056+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.057+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.076+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.081+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.081+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.082+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.082+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.097+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.103+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.103+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.104+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.104+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.118+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.130+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.130+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.137+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.175+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.193+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.193+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.210+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.210+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.233+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.234+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.234+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.248+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.266+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.266+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.283+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.283+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.302+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.304+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.304+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.305+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.305+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.320+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.324+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.324+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.326+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.333+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.13.5
2024-04-22T23:02:28.333+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:28.341+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.342+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.342+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.351+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.374+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.374+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.389+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.389+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.411+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.412+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.412+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.422+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.448+0200	INFO	Detected OS: ubuntu
2024-04-22T23:02:28.448+0200	INFO	Detecting Ubuntu vulnerabilities...
2024-04-22T23:02:28.456+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.490+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.497+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.497+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.501+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.507+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.13.5
2024-04-22T23:02:28.507+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:28.518+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.520+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.520+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.531+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.533+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.533+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.545+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.547+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.547+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.548+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.548+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.570+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.596+0200	INFO	Detected OS: ubuntu
2024-04-22T23:02:28.596+0200	INFO	Detecting Ubuntu vulnerabilities...
2024-04-22T23:02:28.607+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.631+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.632+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.632+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.649+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.654+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.654+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.657+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.664+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.13.5
2024-04-22T23:02:28.664+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:28.675+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.676+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.677+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.691+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.694+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.694+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.712+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.715+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.715+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.716+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.717+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.737+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.771+0200	INFO	Detected OS: debian
2024-04-22T23:02:28.771+0200	INFO	Detecting Debian vulnerabilities...
2024-04-22T23:02:28.855+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.855+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.896+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.901+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.901+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.902+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.913+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.915+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.915+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.929+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.933+0200	INFO	Detected OS: alpine
2024-04-22T23:02:28.933+0200	INFO	Detecting Alpine vulnerabilities...
2024-04-22T23:02:28.938+0200	INFO	Number of language-specific files: 0
2024-04-22T23:02:28.944+0200	WARN	This OS version is no longer supported by the distribution: alpine 3.13.5
2024-04-22T23:02:28.944+0200	WARN	The vulnerability detection may be insufficient because security updates are not provided
2024-04-22T23:02:28.954+0200	INFO	Vulnerability scanning is enabled
2024-04-22T23:02:28.956+0200	INFO	Number of language-specific files: 1
2024-04-22T23:02:28.956+0200	INFO	Detecting gobinary vulnerabilities...
2024-04-22T23:02:28.980+0200	INFO	Vulnerability scanning is enabled
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5b0262091c46]

goroutine 429 [running]:
github.com/aquasecurity/trivy/pkg/sbom/spdx.(*SPDX).isTrivySBOM(...)
	github.com/aquasecurity/trivy@v0.50.0/pkg/sbom/spdx/unmarshal.go:258
github.com/aquasecurity/trivy/pkg/sbom/spdx.(*SPDX).unmarshal(0xc001b5a288, 0xc00076e780)
	github.com/aquasecurity/trivy@v0.50.0/pkg/sbom/spdx/unmarshal.go:72 +0x26
github.com/aquasecurity/trivy/pkg/sbom/spdx.(*SPDX).UnmarshalJSON(0xc001b5a288, {0xc000b91800, 0x17b, 0x200})
	github.com/aquasecurity/trivy@v0.50.0/pkg/sbom/spdx/unmarshal.go:65 +0x20e
encoding/json.(*decodeState).object(0xc00076e668, {0x5b02659a8860?, 0xc001b5a288?, 0xc00137f1d0?})
	encoding/json/decode.go:604 +0x6d0
encoding/json.(*decodeState).value(0xc00076e668, {0x5b02659a8860?, 0xc001b5a288?, 0xc00137f220?})
	encoding/json/decode.go:374 +0x3e
encoding/json.(*decodeState).unmarshal(0xc00076e668, {0x5b02659a8860?, 0xc001b5a288?})
	encoding/json/decode.go:181 +0x11e
encoding/json.(*Decoder).Decode(0xc00076e640, {0x5b02659a8860, 0xc001b5a288})
	encoding/json/stream.go:73 +0x15a
github.com/aquasecurity/trivy/pkg/sbom.Decode({_, _}, {_, _})
	github.com/aquasecurity/trivy@v0.50.0/pkg/sbom/sbom.go:225 +0x6c2
github.com/aquasecurity/trivy/pkg/fanal/analyzer/sbom.sbomAnalyzer.Analyze({}, {0xc001e84cf0?, 0xc001e84cf0?}, {{0x0, 0x0}, {0xc0036960a0, 0x19}, {0x5b0265b41158, 0xc0015d35e0}, {0x7d27c0b38520, ...}, ...})
	github.com/aquasecurity/trivy@v0.50.0/pkg/fanal/analyzer/sbom/sbom.go:39 +0x105
github.com/aquasecurity/trivy/pkg/fanal/analyzer.AnalyzerGroup.AnalyzeFile.func1({0x5b0265b35c00?, 0x5b02683dd340}, {0x5b0265b33af0?, 0xc002024030?})
	github.com/aquasecurity/trivy@v0.50.0/pkg/fanal/analyzer/analyzer.go:430 +0x23a
created by github.com/aquasecurity/trivy/pkg/fanal/analyzer.AnalyzerGroup.AnalyzeFile in goroutine 578
	github.com/aquasecurity/trivy@v0.50.0/pkg/fanal/analyzer/analyzer.go:425 +0x4ab

```

### Expected behavior

not crashing :)

**Comment by nvtkaszpir** [comment]
remvoving `"cve"` key and its content solves crashing, but of course disables cve scans.

**Comment by rchincha** [comment]
@nvtkaszpir is this reproducible for you? it is coming up fine without crashing.

**Comment by nvtkaszpir** [comment]
It may be related to the existing metadata or data of the images, I can provide you with the data if you want (form a snapshot):
```text
/tank/zot # du -sh *
411K	cache.db
2.3G	docker.io
70M	gcr.io
475M	ghcr.io
304M	k8s.gcr.io
923K	meta.db
319M	nvcr.io
8.7G	quay.io
626M	registry.k8s.io
758M	_trivy
```
16GB in total

**Comment by andaaron** [comment]
Hi @nvtkaszpir,

Can you also send us the last entries of the zot logs (/var/log/zot/zot.log)? From what I see it scanned successfully several images, and at some point, probably on a very specific image, it errored. That image could theoretically be identified based on the logs generating when scanning. 

Also did you encounter this error while triggering a CVE scan yourself (using the zli or ui), or has it happened while it was scanning images in the background? That may also help in identifying what images were scanned.

**Comment by nvtkaszpir** [comment]
I'll have to check the logs, I guess it started to happen on its own, because the scan was running on its own every 168h.
Now it started to happen instantly on start because scan time expired and thus it ends in crash loop under systemd service auto restart.

**Comment by andaaron** [comment]
It's probably a newer image which was pushed to the upstream recently and was pulled by zot sync.

**Comment by nvtkaszpir** [comment]
Oh nice.
Will try to do it in the evening, something 12h from now.

**Comment by nvtkaszpir** [comment]
```shell
bin/zot-linux-amd64 --version
{"level":"info","distribution-spec":"1.1.0","commit":"v2.0.3-10-g7b1fc04","binary-type":"-imagetrust-lint-metrics-mgmt-profile-scrub-search-sync-ui-userprefs","go version":"go1.22.1","time":"2024-04-23T20:36:05+02:00","message":"version"}

```
It works, no crashes, after service restart it rescanned all images it could.

**Comment by rchincha** [comment]
@nvtkaszpir thanks for confirming, we will cut a maintenance release for this. 
Close this issue?

**Comment by nvtkaszpir** [comment]
fixed in #2399
